crates/matrix-sdk/src/matrix_auth/mod.rs

```diff
@@ -537,15 +537,15 @@ impl MatrixAuth {
     /// # };
     /// ```
     #[instrument(skip_all)]
-    pub async fn register(
-        &self,
-        request: register::v3::Request,
-    ) -> HttpResult<register::v3::Response> {
+    pub async fn register(&self, request: register::v3::Request) -> Result<register::v3::Response> {
         let homeserver = self.client.homeserver();
         info!("Registering to {homeserver}");
-        self.client.send(request, None).await
+        let response = self.client.send(request, None).await?;
+        if let Ok(session) = MatrixSession::try_from(&response) {
+            self.set_session(session).await.expect("Registration sets token");
```

Comment:
Either the error should be bubbled up, or it should be ignored, but we shouldn't panic here. I think I prefer the latter, because registration in itself worked, and the new contract for the method will be to check whether we've been logged in anyways.